backtrace-library/src/main/java/backtraceio/library/models/json/BacktraceReport.java

```diff
@@ -167,40 +162,53 @@ public BacktraceReport(
      * Create new instance of Backtrace report to send a report
      * with application exception, attributes and attachments
      *
-     * @param exception       current exception
+     * @param throwable       current throwable
      * @param attributes      additional information about application state
      * @param attachmentPaths path to all report attachments
      */
-    public BacktraceReport(
-            Exception exception,
-            Map<String, Object> attributes,
-            List<String> attachmentPaths) {
-
+    public BacktraceReport(Throwable throwable, Map<String, Object> attributes, List<String> attachmentPaths) {
         this.attributes = CollectionUtils.copyMap(attributes);
         this.attachmentPaths = CollectionUtils.copyList(attachmentPaths);
-        this.exception = this.prepareException(exception);
+        this.originalException = throwable;
+        this.exception = this.prepareException(throwable);
+        if (throwable != null) {
+            this.classifier = getExceptionClassifier(throwable);
+        }
         this.exceptionTypeReport = exception != null;
         this.diagnosticStack = new BacktraceStackTrace(exception).getStackFrames();
 
-        if (this.exceptionTypeReport && exception != null) {
-            this.classifier = getExceptionClassifier(exception);
-        }
         this.setDefaultErrorTypeAttribute();
     }
 
-    public String getExceptionClassifier(Exception exception) {
-        if (exception instanceof UnhandledThrowableWrapper) {
-            return ((UnhandledThrowableWrapper) exception).getClassifier();
+    /**
+     * Concat two dictionaries with attributes
+     *
+     * @param report     current report
+     * @param attributes attributes to concatenate
+     * @return concatenated map of attributes from report and from passed attributes
+     */
+    public static Map<String, Object> concatAttributes(
+            BacktraceReport report, Map<String, Object> attributes) {
+        Map<String, Object> reportAttributes = report.attributes != null ? report.attributes :
+                new HashMap<String, Object>();
+        if (attributes == null) {
+            return reportAttributes;
         }
+        reportAttributes.putAll(attributes);
+        return reportAttributes;
+    }
+
+    public String getExceptionClassifier(Throwable exception) {
```

Comment:
I think it can be private

Reply:
Changed - side note: this is leftover from the previous version